### ISSUE TITLE
Training starting from loaded state dict

### ISSUE THREAD

**Issue by gigumay** [ultralytics/ultralytics#19709]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi, 
I have two questions:
1. I would like to intialize a YOLOv8 detection model based on a state dict obtained from another YOLOv8 model. To do so, I have tried the following approach:

```
mdl1 = YOLO("yolov8n.yaml")
mdl1.train()
mdl1_state_dict = mdl1.state_dict()

mdl2 = YOLO("yolov8n.yaml")
mdl2.load_state_dict(mdl1_state_dict)
mdl2.train()
```

However, stepping through the code revealed that `mdl2.train()` calls the `get_model()` function of the `DetectionTrainer` class, which is going to initialize the weights from scratch regardless of whether or not a state dict was loaded beforehand. I gather from some other issues opened here that a potential workaround could be to save the state dict of `mdl1` to a `.pt` file, and to load the latter into `mdl2`. Is this the only solution, or can I start training from a state dict without writing it to a file first. And do I then have to set `resume=True`?

2. My second question is whether model initialisation is deterministic - i.e., all new models are initialised with the same weights - and if not, how I can make sure that it is. I see that by default `mdl.train()` sets the random seed to 0, and since model weights are initialised within `mdl.train()`, I would assume that model initialisation is indeed deterministic, but I wanted to make sure.

Thanks a lot

**Comment by UltralyticsAssistant** [comment]
👋 Hello @gigumay, thank you for reaching out and sharing your detailed query with Ultralytics 🚀! 

We recommend exploring our [Documentation](https://docs.ultralytics.com/) for additional resources, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples, which might help address your questions.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can better assist you in debugging the issue.

For your ❓ Questions:
1. Please share additional details about your use case, such as any modifications you have made, training logs, or relevant configurations. This will help us provide a more precise response.
2. If you are concerned about determinism, you can verify by checking the code and ensuring all seeds and environment variables are properly set. However, feel free to clarify your findings further for our team to assist.

## Upgrade

Make sure you're working with the latest version of Ultralytics. Upgrade with the following command:

```bash
pip install -U ultralytics
```

## Environments

YOLO models can be run in the following verified environments, which come pre-installed with all necessary dependencies (e.g., Python, PyTorch, CUDA):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Community Support

We encourage you to join our growing Ultralytics community for more discussions and support:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time chat
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge and explore ideas

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. 

An Ultralytics engineer will review and assist you with your inquiry shortly. Thank you for your patience and for being part of the Ultralytics community! 🚀

**Comment by Y-T-G** [comment]
You can check this

https://github.com/ultralytics/ultralytics/issues/17135#issuecomment-2435109331

**Comment by gigumay** [comment]
Thanks @Y-T-G. I had thought of this but was wondering if there is a less hacky way - perhaps some functions I am unaware of. Doesn't look like that atm
